geode-lucene/src/upgradeTest/java/org/apache/geode/cache/lucene/RollingUpgradeQueryReturnsCorrectResultsAfterClientAndServersAreRolledOverAllBucketsCreated.java

```diff
@@ -31,6 +32,7 @@
 public class RollingUpgradeQueryReturnsCorrectResultsAfterClientAndServersAreRolledOverAllBucketsCreated
     extends LuceneSearchWithRollingUpgradeDUnit {
 
+  @Ignore
```

Comment:
Can you comment on why this test is being ignored, and when or how we will re-write or re-enable it again?

Reply:
The main sequence in this test that causes the failure is:


1. start two servers with old version using Lucene 6
2. roll one server to new version server using Lucene 7
3. do puts into primary buckets in new server which creates entries in the fileAndChunk region with Lucene 7 format
4. stop the new version server which causes the old version server to become primary for those buckets
5. do a query which causes the IndexFormatTooNewException to be thrown

So the old server cannot read the new index format. This test can be valid for the version of the Geode which is using Lucene without changes in index format.

Reply:
If it's not possible to rewrite this test to allow it to pass in all cases, it would be better that it simply be removed. However, it seems like it should be doable to add some logic to determine the expected behaviour of the query based on the version of Geode under test (and hence the version of Lucene).

Reply:
What did you mean by this logic to determine the expected behaviour of the query based on the Geode version?  In versions before Lucene is upgraded, the test(always doing the upgrade to the newest version) is not changed and it works. Maybe you think that we need this change for the future version of Geode when Lucene format will be the same, and test between that versions should work as before this changes? Or you mean something else?

Reply:
Yeah, I think it would be good to have two paths for the test. One where we know there is a Lucene version mismatch, where we assert that the query fails or throws (or whatever the expected behaviour ends up being in that situation) and one where we know that the Lucene version is the same on both old and new servers (like after 1.15 is released with the Lucene version changed, and we can start rolling from 1.15 to 1.16) where we assert that the query succeeds.